File: Src/Backend/SourceCode/TEK.SeatPlan.Entity/Employee.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace TEK.SeatPlan.Entity
{
	public class Employee : TrackedEntity
	{
		[Required]
		[StringLength(55)]
		public string FirstName { get; set; }

		[Required]
		[StringLength(55)]
		public string LastName { get; set; }

		[Required]
		[StringLength(55)]
		public string Email { get; set; }

		[StringLength(255)]
		public string Description { get; set; }

		[Required]
		public bool Active { get; set; }

		public virtual Project Project { get; set; }

		public virtual Seat Seat { get; set; }

		public virtual EmployeeStatus Status { get; set; }

		public virtual ICollection<SeatChangeLog> SeatChangeLog { get; set; }
	}
}

File: Src/Frontend/src/app/core/models/employee.ts
import {Project} from './project';
import {Status} from './status';
import {Seat} from './seat';

export class Employee {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  description: string;
  project: Project;
  status: Status;
  seat: Seat;

  constructor(project: Project, status: Status, seat: Seat) {
    this.id = undefined;
    this.firstName = '';
    this.lastName = '';
    this.email = '';
    this.description = '';
    this.project = project;
    this.status = status;
    this.seat = seat;
  }
}
